test/functional/api/v2/discovered_hosts_controller_test.rb

```diff
@@ -75,17 +77,28 @@ def test_auto_provision_success
     disable_orchestration
     facts = @facts.merge({"somefact" => "abc"})
     host = Host::Discovered.import_host_and_facts(facts).first
-    FactoryGirl.create(:discovery_rule, :priority => 1, :search => "facts.somefact = abc", :hostgroup => hostgroups(:common))
+    FactoryGirl.create(:discovery_rule, :priority => 1, :search => "facts.somefact = abc", :hostgroup => hostgroups(:common),
+                       :organizations => [host.organization], :locations => [host.location])
     post :auto_provision, { :id => host.id }
     assert_response :success
   end
 
-  def test_auto_provision_success_and_delete
+  def test_auto_provision_with_wrong_org_or_loc_fail
     disable_orchestration
     facts = @facts.merge({"somefact" => "abc"})
     host = Host::Discovered.import_host_and_facts(facts).first
     FactoryGirl.create(:discovery_rule, :priority => 1, :search => "facts.somefact = abc", :hostgroup => hostgroups(:common))
     post :auto_provision, { :id => host.id }
+    assert_response 200
+  end
```

Comment:
This can only be tested correctly after: https://github.com/theforeman/foreman_discovery/pull/171 is merged. For now, it just succeeds with no error